rabbitmq_queue_master_balancer/rabbitmq_queue_master_balancer.py

```diff
@@ -0,0 +1,209 @@
+
+parser.add_argument(
+    '--api-url',
+    type=str,
+    default='localhost:15672'
+)
+
+parser.add_argument(
+    '--user',
+    type=str,
+    default='guest'
+)
+
+parser.add_argument(
+    '--password',
+    type=str,
+    default='guest'
+)
+
+parser.add_argument(
+    '-L', '--log-level', help='Log level',
+    default='info',
+    choices=(
+        'critical', 'fatal', 'error', 'warning',
+        'warn', 'info', 'debug', 'notset'
+    ),
+)
+
+parser.add_argument(
+    '--log-format', choices=LogFormat.choices(), default=LogFormat.stream
+)
+
+parser.add_argument(
+    '--dry-run',
+    action="store_true",
+    help='Show how many queues will move from one node (with a maximum queues) \
+    to another (with a minimum queues)'
+)
+
+parser.add_argument(
+    '--queue-delta',
+    type=int,
+    default=3,
+    help='Reasonable delta of queues between max and min numbers of queues \
+    on node when script do nothing'
+)
+
+parser.add_argument(
+    '--relocate-queues',
+    type=int,
+    default=2,
+    help='How many queues will relocate in one time'
+)
+
+arguments = parser.parse_args()
+
+log = logging.getLogger()
+
+
+def wait_for_client(client):
+    '''
+    :param client:
+    :return: bool
+    '''
+
+    log.debug('RabbitMQ http not ready, waiting...')
+    try:
+        return client.is_alive()
+    except NetworkError:
+        time.sleep(5)
+        wait_for_client(client)
+
+
+def nodes_dict(nodes_info_data, vhost_names: list):
+    '''
+    :param client
+    :param vhost_names list of vhosts
+    :return: dict: Key is a name of rabbit node, Value is empty list
+    '''
+    nodes_dictionary = dict.fromkeys(
+        (node['name'] for node in nodes_info_data), dict.fromkeys(
+            (vhost for vhost in vhost_names)
+        )
+    )
+    return nodes_dictionary
+
+
+def master_nodes_queues(nodes_dict: dict, vhost_names: list):
+    '''
+    :param queues_data: list of dicts with all queues info
+    :param vhost_names list of vhosts
+    :return: dict {node_name: {vhost1: list_queues, vhost2: list_queues}
+    '''
+
+    master_nodes_queues_dict = {}
+
+    queues_data = client.get_queues()
+    for node in nodes_dict.items():
+
+        vhost_dict = {}
+
+        for vhost in vhost_names:
+            list_queues = []
+
+            for queue in queues_data:
+                if node[0] == queue['node'] and queue['vhost'] == vhost:
+                    list_queues.append(queue['name'])
+
+            vhost_dict.update({vhost: list_queues})
+
+            master_nodes_queues_dict.update({node[0]: vhost_dict})
+
+    return master_nodes_queues_dict
+
+
+def calculate_queues(master_nodes_queues_dict: dict):
+    '''
+    :param master_nodes_queues_dict: dict
+    :return: dict {node1: number_queues, node2: number_queues,}
+    '''
+
+    calculated_dict = {}
+
+    for node in master_nodes_queues_dict.items():
+        counter = 0
+        for i in node[1].values():
+            counter += len(i)
```

Comment:
`counter = sum([i for i in vhost.values()])`

Reply:
counter = sum([len(i) for i in vhost.values()])

Reply:
```python
counter = sum(map(len, vhost.values()))
```